sky/cli.py

```diff
@@ -1568,7 +1568,15 @@ def status(all: bool, refresh: bool, show_spot_jobs: bool, clusters: List[str]):
             click.echo(f'\n{colorama.Fore.CYAN}{colorama.Style.BRIGHT}'
                        f'Managed spot jobs{colorama.Style.RESET_ALL}')
             with log_utils.safe_rich_status('[cyan]Checking spot jobs[/]'):
-                num_in_progress_jobs, msg = spot_jobs_future.get()
+                try:
+                    num_in_progress_jobs, msg = spot_jobs_future.get()
+                except KeyboardInterrupt:
+                    pool.terminate()
+                    # Set to -1, so that the controller is not considered
+                    # down, and the hint for showing sky spot queue
+                    # will still be shown.
+                    num_in_progress_jobs = -1
+                    msg = 'KeyboardInterrupt'
```

Comment:
I still see the stacktrace, and a duplicate line showing `KeyboardInterrupt`. Maybe fine to leave this fix to future, if too difficult to handle.

<img width="1420" alt="Screen Shot 2023-03-10 at 11 18 48" src="https://user-images.githubusercontent.com/592670/224406673-8d633f0e-0f23-474d-9b4b-2e8cd7a11981.png">

Reply:
Weirdly, I cannot reproduce the problem shown in the figure, but the keyboard interruption handling here is indeed a bit tricky as it involves multiple processes. Let's leave it for a future PR. : )